src/core/thread/network_data_tlvs.hpp

```diff
@@ -890,10 +914,7 @@ class BorderRouterEntry : public Equatable<BorderRouterEntry>
      * @returns the Preference value.
      *
      */
-    int8_t GetPreference(void) const
-    {
-        return static_cast<int8_t>(static_cast<int16_t>(HostSwap16(mFlags)) >> kPreferenceOffset);
```

Comment:
I copied this into a new function `PreferenceFromFlags()` in this PR.

But I just noticed that we are using right shift of a signed int. I think his is not explicitly specfiied by C++ standard and its behavior may be implementation/toolchain dependent. Better to avoid it if we can.

I will address it in a separate PR (into `main`) and then we can rebase and fix here as well.

Reply:
This is what I found: https://en.cppreference.com/w/cpp/language/operator_arithmetic

> For negative a, the behavior of a << b is undefined.
> 
> For unsigned a and for signed and non-negative a, the value of a >> b is the integer part of a/2b
> 
> For negative a, the value of a >> b is implementation-defined (in most implementations, this performs arithmetic right shift, so that the result remains negative).
>